idx/memory/tag_query.go

```diff
@@ -49,37 +71,66 @@ func (a KvReByCost) Len() int           { return len(a) }
 func (a KvReByCost) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
 func (a KvReByCost) Less(i, j int) bool { return a[i].cost < a[j].cost }
 
+//TagQuery runs a set of pattern or string matches on tag keys and values against
+// the index. It is executed via:
+// Run() which returns a set of matching MetricIDs
+// RunGetTags() which returns the set of tag values found amongst the matching MetricIDs
 type TagQuery struct {
-	from      int64
-	equal     []kv
-	match     []kvRe
-	notEqual  []kv
-	notMatch  []kvRe
-	startWith int
+	from        int64
+	equal       []kv
+	match       []kvRe
+	notEqual    []kv
+	notMatch    []kvRe
+	prefix      []kv
+	startWith   int // expression type to generate the initial result set from
+	filterTagBy int // if tags should be filtered by prefix or regex match (max one)
```

Comment:
this filterTagBy value seems to be used along with the two tagMatch and tagPrefix fields below, and don't seem to have anything to do with the fields its currently listed with.  moving it down and documenting when it is used would help a lot with understandability

Reply:
https://github.com/grafana/metrictank/pull/779/commits/54665ad4207e1f102377c47cf3f127e28c4a0b88